src/tools/idlc/src/plugin.c

```diff
@@ -84,8 +143,38 @@ idlc_load_generator(idlc_generator_plugin_t *plugin, const char *lang)
     return 0;
   }
 
+  /* special case for python generator
+        The 'active' idlpy library is dependend on which python is active. Idlpy is installed
+        as part of the cyclonedds python package and it can very well be that multiple installations
+        are present on the system, by use of virtual environments, user installs and system installs.
+        Sadly activating a python distribution does not set a any library loading paths. However,
+        the python cyclonedds package has a __idlc__ module that prints the path when executed. The
+        'python3' executable is always guaranteed to point to the active python so we always get the
+        correct idlpy library.
+   */
+  if (idl_strcasecmp(lang, "py") == 0) {
+    int retcode;
+    if (run_library_locator("python3" SUBPROCESS_EXEC " -m cyclonedds.__idlc__", &file, &retcode) != 0) {
```

Comment:
According to the MSDN page `_popen` uses the command processor and executes the command as the first command in that processor. Probably easier to remove `SUBPROCESS_EXEC`. My reasoning here being that it may actually cause problems when executed in Cygwin shells etc(?)